include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
>BS. boilerplate - fuck no.

I take a hardline stance on macros. I will reject pull requests that use them. I've been pretty successful at making a codebase that's accessible to new contributors so far.

>I don't understand how a char* enum_strings[] can "use anonymous structs". We're not talking about generic data structure here. And remember that calling the function means the compiler type-checks for this.

I was talking about the generalized solution for getting the length of an array in items. It doesn't necessarily bring any significant benefits to the table in this case but it works for more situations, and if one solution works in many places then it should be preferred over a solution that only works for a few.